src/cpp/rtps/builtin/discovery/participant/PDPSimpleListener.cpp

```diff
@@ -149,6 +149,8 @@ void PDPSimpleListener::onNewCacheChangeAdded(RTPSReader* reader, const CacheCha
             {
                 listener->onParticipantDiscovery(this->mp_SPDP->getRTPSParticipant()->getUserRTPSParticipant(), std::move(info));
             }
+
+            return; // change already removed from history
```

Comment:
Then, where is it removed?

Reply:
Is is removed inside the successful PDPSimple::removeRemoteParticipant call.